format/puffin-spec.md

```diff
@@ -123,6 +123,54 @@ The blob metadata for this blob may include following properties:
 
 - `ndv`: estimate of number of distinct values, derived from the sketch.
 
+#### `delete-vector-v1` blob type
+
+A serialized delete vector (bitmap) that represents the positions of rows in a
+file that are deleted.  A set bit at position P indicates that the row at
+position P is deleted.
+
+The vector supports positive 64-bit positions (the most significant bit must be
+0), but is optimized for cases where most positions fit in 32 bits by using a
+collection of 32-bit Roaring bitmaps.  64-bit positions are divided into a
+32-bit "key" using the most significant 4 bytes and a 32-bit sub-position using
+the least significant 4 bytes. For each key in the set of positions, a 32-bit
+Roaring bitmap is maintained to store a set of 32-bit sub-positions for that
+key.
+
+To test whether a certain position is set, its most significant 4 bytes (the
+key) are used to find a 32-bit bitmap and the least significant 4 bytes (the
+sub-position) are tested for inclusion in the bitmap. If a bitmap is not found
+for the key, then it is not set.
+
+The serialized blob contains:
+* The length of the vector and magic bytes stored as 4 bytes, big-endian
+* A 4-byte magic sequence, `D1 D3 39 64`
+* The vector, serialized as described below
+* A CRC-32 checksum of the serialized vector as 4 bytes, big-endian
+
+The position vector is serialized using the Roaring bitmap
+["portable" format][roaring-bitmap-portable-serialization]. This representation
+consists of:
+
+* The number of 32-bit Roaring bitmaps, serialized as 8 bytes, little-endian
+* For each 32-bit Roaring bitmap, ordered by unsigned comparison of the 32-bit keys:
+    - The key stored as 4 bytes, little-endian
+    - A [32-bit Roaring bitmap][roaring-bitmap-general-layout]
+
+Note that the length and CRC fields are stored using big-endian, but the
+Roaring bitmap format uses little-endian values. Big endian values were chosen
+for compatibility with existing deletion vectors.
+
+The blob metadata must include the following properties:
+
+* `referenced-data-file`: location of the data file the delete vector applies
```

Comment:
Would it make sense to include min row and max row in the delete vector here (we can avoid doing any more work if the row-group being processed is outside this range).  This might also be cheap enough to extract from the data itself, so it might be a non-issue.

Reply:
The problem is that it will be in the footer, so it will require an extra request.
Are there good use cases in mind, @emkornfield?

Reply:
I was assuming footer parsing first.  I think if the intent to jump directly to a location from manifests, then it would serve no purpose (I added some questions in this regard on the table spec changes).

Reply:
One follow-up question if we aren't planning on reading the footer, why are we storing cardinality?

Reply:
The cardinality is part of making these delete files self-describing and is up for discussion. I can image some maintenance operations compacting DV files without touching the rest of the metadata.

Reply:
How will it be consumed in general though?  Is this another sanity check value?  Or is there a specific use for it?

Reply:
Cardinality is also stored in the Iceberg metadata entry as record count.